src/main/webapp/app/components/newCompanyForm/NewCompanyForm.tsx

```diff
@@ -0,0 +1,333 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import { observable, action, computed } from 'mobx';
+import {
+  AvField,
+  AvForm,
+  AvGroup,
+  AvInput,
+  AvFeedback,
+} from 'availity-reactstrap-validation';
+import { Button, Col, Row } from 'react-bootstrap';
+import {
+  LicenseType,
+  LICENSE_TITLES,
+  LicenseStatus,
+  CompanyType,
+  COMPANY_TYPE_TITLES,
+  LICENSE_STATUS_TITLES,
+  LicenseModel,
+  LICENSE_MODEL_TITLES,
+} from 'app/config/constants';
+import { getSectionClassName } from 'app/pages/account/AccountUtils';
+import autobind from 'autobind-decorator';
+import { CompanyVM } from 'app/shared/api/generated/API';
+import styles from './styles.module.scss';
+import Select from 'react-select';
+
+type INewCompanyFormProps = {
+  onValidSubmit: (newCompany: Partial<CompanyVM>) => void;
+};
+
+type IDomainListBoxProps = {
+  domainList: string[];
+  updateDomainList: (actionType: DomainListActions, event: any) => void;
+};
+
+enum DomainListActions {
+  ADD,
+  DELETE,
+}
+
+@observer
+class DomainListBox extends React.Component<IDomainListBoxProps> {
+  render() {
+    return (
+      <div
+        style={{
+          border: '1px solid #ced4da',
+          borderRadius: '0.25rem',
+          padding: '0.375rem 0.75rem',
+          overflowY: 'scroll',
+          minHeight: '100px',
+          maxHeight: '100px',
+          marginTop: '28px',
+        }}
+      >
+        <div className={styles.main}>
+          {this.props.domainList.map(domain => (
+            <div key={domain} className={styles.content_container}>
+              <span
+                className={styles.delete}
+                onClick={() =>
+                  this.props.updateDomainList(DomainListActions.DELETE, domain)
+                }
+              >
+                <i className="fa fa-times-circle"></i>
+              </span>
+              <span style={{ margin: '5px 10px 5px 5px' }}>{domain}</span>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+export type IFormTextAreaProps = {
+  onTextAreaChange: (event: any) => void;
+  label: string;
+  style?: React.CSSProperties;
+};
+
+export const FormTextAreaField: React.FunctionComponent<IFormTextAreaProps> = props => {
+  return (
+    <div className="form-group">
+      <label className="form-label">{props.label}</label>
+      <textarea
+        onChange={props.onTextAreaChange}
+        style={props.style || { minHeight: '50px' }}
+        className="form-control"
+      ></textarea>
+    </div>
+  );
+};
+
+export type IFormSelectWithLabelProps = {
+  onSelection: (selectedOption: any) => void;
+  labelText: string;
+  name: string;
+  defaultValue: { value: string; label: string };
+  options: { value: string; label: string }[];
+};
+
+export const COMPANY_FORM_OPTIONS = {
+  companyType: Object.keys(CompanyType).map(type => {
+    return {
+      value: CompanyType[type],
+      label: COMPANY_TYPE_TITLES[CompanyType[type]],
```

Comment:
If we are looping through the enum and get the title from `COMPANY_TYPE_TITLES`, then the type of `COMPANY_TYPE_TITLES` should be `{ [key in CompanyType]: string }`, the seem goes to other const values.